zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/ElasticsearchStorage.java

```diff
@@ -56,40 +77,41 @@
     List<String> get();
   }
 
-  static final MediaType APPLICATION_JSON = MediaType.parse("application/json");
-
-  public static Builder newBuilder(OkHttpClient client) {
-    return new $AutoValue_ElasticsearchStorage.Builder()
-        .client(client)
-        .hosts(Collections.singletonList("http://localhost:9200"))
-        .maxRequests(64)
-        .strictTraceId(true)
-        .searchEnabled(true)
-        .index("zipkin")
-        .dateSeparator('-')
-        .indexShards(5)
-        .indexReplicas(1)
-        .namesLookback(86400000)
-        .shutdownClientOnClose(false)
-        .flushOnWrites(false)
-        .autocompleteKeys(Collections.emptyList())
-        .autocompleteTtl((int) TimeUnit.HOURS.toMillis(1))
-        .autocompleteCardinality(5 * 4000); // Ex. 5 site tags with cardinality 4000 each
-  }
-
   public static Builder newBuilder() {
-    Builder result = newBuilder(new OkHttpClient());
-    result.shutdownClientOnClose(true);
-    return result;
+    return new $AutoValue_ElasticsearchStorage.Builder()
+      .clientCustomizer(unused -> {})
+      .clientFactoryCustomizer(unused -> {})
+      .hosts(Collections.singletonList("http://localhost:9200"))
+      .maxRequests(64)
+      .strictTraceId(true)
+      .searchEnabled(true)
+      .index("zipkin")
+      .dateSeparator('-')
+      .indexShards(5)
+      .indexReplicas(1)
+      .namesLookback(86400000)
+      .flushOnWrites(false)
+      .autocompleteKeys(Collections.emptyList())
+      .autocompleteTtl((int) TimeUnit.HOURS.toMillis(1))
+      .autocompleteCardinality(5 * 4000); // Ex. 5 site tags with cardinality 4000 each
   }
 
   abstract Builder toBuilder();
 
   @AutoValue.Builder
   public abstract static class Builder extends StorageComponent.Builder {
-    abstract Builder client(OkHttpClient client);
+    /**
+     * Customizes the {@link HttpClientBuilder} used when connecting to ElasticSearch. Mostly for
+     * testing.
+     */
+    public abstract Builder clientCustomizer(Consumer<HttpClientBuilder> clientCustomizer);
```

Comment:
if only for testing, we should make it package private like the above. the more we expose the more we break on next refactor :P

Reply:
I suspect one or two of these are exposed for zipkin-aws, if so a comment under the javadoc saying literally what is using it will help us from accidentally breaking.